src/shared/sgxlkl_enclave_config.c

```diff
@@ -0,0 +1,560 @@
+                i = parser->path[parser->depth - 2].index;
+                if (i >= 8)
+                    FAIL("too many values for clock_res");
+                memcpy(&cfg->clock_res[i].resolution, un->string, 17);
+            });
+
+            JPATHT("mode", JSON_TYPE_STRING, {
+                cfg->mode = string_to_sgxlkl_enclave_mode_t(un->string);
+            });
+            JBOOL("fsgsbase", cfg->fsgsbase);
+            JBOOL("verbose", cfg->verbose);
+            JBOOL("kernel_verbose", cfg->kernel_verbose);
+            JSTRING("kernel_cmd", cfg->kernel_cmd);
+            JSTRING("sysctl", cfg->sysctl);
+            JBOOL("swiotlb", cfg->swiotlb);
+
+            JSTRING("cwd", cfg->cwd);
+            JPATHT("args", JSON_TYPE_STRING, {
+                strdupz(&cfg->args[i], un->string);
+            });
+            JPATHT("env", JSON_TYPE_STRING, {
+                strdupz(&cfg->env[i], un->string);
+            });
+            JU64("auxv.a_type", cfg->auxv[i].a_type);
+            JU64("auxv.a_val", cfg->auxv[i].a_un.a_val);
+            JPATHT("host_import_env", JSON_TYPE_STRING, {
+                strdupz(&cfg->host_import_env[i], un->string);
+            });
+
+            JPATHT("exit_status", JSON_TYPE_STRING, {
+                cfg->exit_status =
+                    string_to_sgxlkl_exit_status_mode_t(un->string);
+            });
+
+            JHEXBUF("root.key", data->config->root.key);
+            JSTRING("root.key_id", data->config->root.key_id);
+            JBOOL("root.readonly", data->config->root.readonly);
+            JSTRING("root.roothash", data->config->root.roothash);
+            JU64("root.roothash_offset", data->config->root.roothash_offset);
+            JBOOL("root.overlay", data->config->root.overlay);
+
+#define MOUNT() (&(cfg->mounts[parser->path[parser->depth - 2].index]))
+            JBOOL("mounts.create", MOUNT()->create);
+            JBOOL("mounts.fresh_key", MOUNT()->fresh_key);
+            JHEXBUF("mounts.key", MOUNT()->key);
+            JSTRING("mounts.key_id", MOUNT()->key_id);
+            if (MATCH("mounts.destination"))
+            {
+                SEEN(parser);
+                size_t len = strlen(un->string);
+                if (len > SGXLKL_DISK_MNT_MAX_PATH_LEN)
+                    FAIL("invalid length of 'mnt' for disk %d\n", i);
+                memcpy(MOUNT()->destination, un->string, len + 1);
+                return JSON_OK;
+            }
+            JBOOL("mounts.readonly", MOUNT()->readonly);
+            JSTRING("mounts.roothash", MOUNT()->roothash);
+            JU64("mounts.roothash_offset", MOUNT()->roothash_offset);
+            JU64("mounts.size", MOUNT()->size);
+
+            sgxlkl_image_sizes_config_t* sizes = &cfg->image_sizes;
+            JU64("image_sizes.num_heap_pages", sizes->num_heap_pages);
+            JU64("image_sizes.num_stack_pages", sizes->num_stack_pages);
+
+            // Else element is unknown. We ignore this to allow newer
```

Comment:
I think so too. I changed it because you said we have to support old images being launched from newer launchers, i.e. there will be settings that the old image doesn't know about.

Reply:
Hm? By image do you mean enclave image? If so, why would you ever have a mismatch?

Reply:
See approximately three other threads... :-) New infrastructure (launcher) + old container (thus old enclave image).

Reply:
I think you misunderstand, the launcher and the OE enclave image are *always* together. You are *never* updating one without the other. The *Docker image* prepared by the user does *neither* contain the launcher *nor* the OE enclave image, it only contains the disk image and configuration files.

Reply:
Yes, same answer as in all the other threads about this. I don't care what the name of the image which contains both is, as long as it always contains both of the same version.

Reply:
Let's make this an error. Using unknown fields is either a typo or the user targeted a newer SGX-LKL that's not deployed.